packages/core/src/State.ts

```diff
@@ -47,13 +55,12 @@ export class State<
 > {
   public value: StateValue;
   public context: TContext;
-  public historyValue: HistoryValue<TContext, TEvent> = {};
+  public historyValue: Readonly<HistoryValue<TContext, TEvent>> = {};
```

Comment:
If we want to validate this kind of stuff at the type-level then this probably should be a `DeepReadonly` (that would have to be added). However, the same applies to most of the things here - what makes `historyValue` to be special that it's the only one wrapped in `Readonly`?

Reply:
I added this to catch some bugs - it was being accidentally mutated.

Reply:
What was the bug? Should a test case be added? :p

Reply:
The entire state should be `Readonly`